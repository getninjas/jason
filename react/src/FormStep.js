import React, { Component } from 'react';

class FormStep extends Component {

  state = {
    formData: this.props.valueInput
  }

  render() {
    return (
      <div id={this.props.index}>
        <h2>Step {this.props.index}</h2>

        {this.props.step.map((field, i) => {
          return (
            <div key={i}>
              <label htmlFor={field.name}>{field.label}</label><br />
              <input
                type="text"
                placeholder={field.mask}
                required={field.required}
                value={this.state.formData[field.name]}
                name={field.name}
                onChange={this.props.dataHandler} />
            </div>
          )
        })}

        {this.props.index !== 0 && this.props.stepBy && (
          <button type="button" onClick={this.props.onBackClick}>Back</button>
        )}

        <button>Next</button>
      </div>
    );
  }
}

export default FormStep;
