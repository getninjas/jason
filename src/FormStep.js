import React from 'react';

const FormStep = ({step, index, last}) => {
  return (
    <div id={index}>
      <h2>Tab {index}</h2>

      {step.map((field, i) => {
        return (
          <div key={i}>
            <label htmlFor={field.name}>{field.label}</label><br />
            <input type="text" placeholder={field.mask} required={field.required} />
          </div>
        )
      })}

      <button>
        Next
      </button>
    </div>
  );
}

export default FormStep;
