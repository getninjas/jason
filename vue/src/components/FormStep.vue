<template>
  <h1>{{message}}</h1>
</template>

<script>
export default {
  name: 'FormStep',
  data() {
    return {
      message: 'Hello',
    }
  },
};
</script>
