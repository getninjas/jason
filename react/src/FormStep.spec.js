'use stict';

import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import FormStep from './FormStep';

const data = {
  "_embedded": {
  "request_fields": [{
      "reference": null,
      "name": "Qual será o serviço?",
      "label": "Qual será o serviço?",
      "placeholder": "Qual será o serviço?",
      "mask": "tipo de serviço",
      "type": "enumerable",
      "required": true,
      "allow_multiple_value": true,
      "allow_custom_value": false,
      "values": {
          "Coloração": "Coloração",
          "Corte": "Corte",
          "Escova": "Escova",
          "Escova progressiva/definitiva": "Escova progressiva/definitiva",
          "Luzes": "Luzes",
          "Manicure": "Manicure",
          "Pedicure": "Pedicure",
          "Penteado": "Penteado"
      },
      "_embedded": {
          "nested_fields": []
      }
  }, {
      "reference": null,
      "name": "Para quem será o serviço?",
      "label": "Para quem será o serviço?",
      "placeholder": "Para quem será o serviço?",
      "mask": "indique para quem será o serviço",
      "type": "enumerable",
      "required": false,
      "allow_multiple_value": true,
      "allow_custom_value": false,
      "values": {
          "Criança": "Criança",
          "Homem": "Homem",
          "Mulher": "Mulher"
      },
      "_embedded": {
          "nested_fields": []
      }
  }, {
      "reference": null,
      "name": "O serviço será para quantas pessoas?",
      "label": "O serviço será para quantas pessoas?",
      "placeholder": "O serviço será para quantas pessoas?",
      "mask": "indique o número de pessoas",
      "type": "enumerable",
      "required": false,
      "allow_multiple_value": false,
      "allow_custom_value": false,
      "values": {
          "1": "1",
          "2": "2",
          "Mais de 3": "Mais de 3"
      },
      "_embedded": {
          "nested_fields": []
      }
  }, {
      "reference": null,
      "name": "Para quando você precisa deste serviço?",
      "label": "Para quando você precisa deste serviço?",
      "placeholder": "Para quando você precisa deste serviço?",
      "mask": "Indique o prazo do serviço",
      "type": "enumerable",
      "required": true,
      "allow_multiple_value": false,
      "allow_custom_value": false,
      "values": {
          "Hoje ou nos próximos dias": "Hoje ou nos próximos dias",
          "Nos próximos 30 dias": "Nos próximos 30 dias",
          "Nos próximos 3 meses": "Nos próximos 3 meses",
          "Ainda não tenho previsão": "Ainda não tenho previsão"
      },
      "_embedded": {
          "nested_fields": []
      }
  }, {
      "name": "Informações Adicionais",
      "label": "Informações Adicionais",
      "type": "big_text",
      "placeholder": "Descreva o que você precisa",
      "required": false
  }],
  "user_fields": [{
      "name": "lat_lng",
      "label": "Em que bairro será o serviço?",
      "type": "lat_lng",
      "placeholder": "",
      "required": true
  }, {
      "name": "name",
      "label": "Nome",
      "type": "small_text",
      "placeholder": "",
      "required": true
  }, {
      "name": "email",
      "label": "Email",
      "type": "email",
      "placeholder": "Ex: nome@email.com",
      "required": true
  }, {
      "name": "phone",
      "label": "Celular",
      "type": "phone",
      "placeholder": "",
      "required": true
  }]
}};

let formData = {}

describe('#Test FormStep Component', () => {
  it('#Should render FormStep for type normal on first step', () => {
    const tree = renderer.create(<FormStep
        step={data._embedded.request_fields}
        index={0}
        valueInput={formData}
        onBackClick={() => {}}
        dataHandler={() => {}}
        stepBy={'normal' === 'stepby' ? true : false}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('#Should render FormStep for type normal on second step', () => {
    const tree = renderer.create(<FormStep
        step={data._embedded.user_fields}
        index={1}
        valueInput={formData}
        onBackClick={() => {}}
        dataHandler={() => {}}
        stepBy={'normal' === 'stepby' ? true : false}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('#Should render FormStep for type stepby on first step', () => {
    const tree = renderer.create(<FormStep
        step={data._embedded.request_fields}
        index={0}
        valueInput={formData}
        onBackClick={() => {}}
        dataHandler={() => {}}
        stepBy={'stepby' === 'stepby' ? true : false}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('#Should render FormStep for type stepby on second step', () => {
    const tree = renderer.create(<FormStep
        step={data._embedded.user_fields}
        index={1}
        valueInput={formData}
        onBackClick={() => {}}
        dataHandler={() => {}}
        stepBy={'stepby' === 'stepby' ? true : false}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('#Should not render back button', () => {
    const wrapper = shallow(<FormStep
      step={data._embedded.request_fields}
      index={0}
      valueInput={formData}
      onBackClick={() => {}}
      dataHandler={() => {}}
      stepBy={'stepby' === 'stepby' ? true : false}
    />);

    expect(wrapper.contains('Back')).toEqual(false);
  });

  it('#Should render back button', () => {
    const wrapper = shallow(<FormStep
      step={data._embedded.user_fields}
      index={1}
      valueInput={formData}
      onBackClick={() => {}}
      dataHandler={() => {}}
      stepBy={'stepby' === 'stepby' ? true : false}
    />);

    expect(wrapper.contains('Back')).toEqual(true);
  });
});
