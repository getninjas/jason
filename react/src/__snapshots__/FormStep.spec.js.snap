// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test FormStep Component #Should render FormStep for type normal on first step 1`] = `
<div
  id={0}
>
  <h2>
    Step 
    0
  </h2>
  <div>
    <label
      htmlFor="Qual será o serviço?"
    >
      Qual será o serviço?
    </label>
    <br />
    <input
      name="Qual será o serviço?"
      onChange={[Function]}
      placeholder="tipo de serviço"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="Para quem será o serviço?"
    >
      Para quem será o serviço?
    </label>
    <br />
    <input
      name="Para quem será o serviço?"
      onChange={[Function]}
      placeholder="indique para quem será o serviço"
      required={false}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="O serviço será para quantas pessoas?"
    >
      O serviço será para quantas pessoas?
    </label>
    <br />
    <input
      name="O serviço será para quantas pessoas?"
      onChange={[Function]}
      placeholder="indique o número de pessoas"
      required={false}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="Para quando você precisa deste serviço?"
    >
      Para quando você precisa deste serviço?
    </label>
    <br />
    <input
      name="Para quando você precisa deste serviço?"
      onChange={[Function]}
      placeholder="Indique o prazo do serviço"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="Informações Adicionais"
    >
      Informações Adicionais
    </label>
    <br />
    <input
      name="Informações Adicionais"
      onChange={[Function]}
      placeholder={undefined}
      required={false}
      type="text"
      value={undefined}
    />
  </div>
  <button>
    Next
  </button>
</div>
`;

exports[`#Test FormStep Component #Should render FormStep for type normal on second step 1`] = `
<div
  id={1}
>
  <h2>
    Step 
    1
  </h2>
  <div>
    <label
      htmlFor="lat_lng"
    >
      Em que bairro será o serviço?
    </label>
    <br />
    <input
      name="lat_lng"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="name"
    >
      Nome
    </label>
    <br />
    <input
      name="name"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <br />
    <input
      name="email"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="phone"
    >
      Celular
    </label>
    <br />
    <input
      name="phone"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <button>
    Next
  </button>
</div>
`;

exports[`#Test FormStep Component #Should render FormStep for type stepby on first step 1`] = `
<div
  id={0}
>
  <h2>
    Step 
    0
  </h2>
  <div>
    <label
      htmlFor="Qual será o serviço?"
    >
      Qual será o serviço?
    </label>
    <br />
    <input
      name="Qual será o serviço?"
      onChange={[Function]}
      placeholder="tipo de serviço"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="Para quem será o serviço?"
    >
      Para quem será o serviço?
    </label>
    <br />
    <input
      name="Para quem será o serviço?"
      onChange={[Function]}
      placeholder="indique para quem será o serviço"
      required={false}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="O serviço será para quantas pessoas?"
    >
      O serviço será para quantas pessoas?
    </label>
    <br />
    <input
      name="O serviço será para quantas pessoas?"
      onChange={[Function]}
      placeholder="indique o número de pessoas"
      required={false}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="Para quando você precisa deste serviço?"
    >
      Para quando você precisa deste serviço?
    </label>
    <br />
    <input
      name="Para quando você precisa deste serviço?"
      onChange={[Function]}
      placeholder="Indique o prazo do serviço"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="Informações Adicionais"
    >
      Informações Adicionais
    </label>
    <br />
    <input
      name="Informações Adicionais"
      onChange={[Function]}
      placeholder={undefined}
      required={false}
      type="text"
      value={undefined}
    />
  </div>
  <button>
    Next
  </button>
</div>
`;

exports[`#Test FormStep Component #Should render FormStep for type stepby on second step 1`] = `
<div
  id={1}
>
  <h2>
    Step 
    1
  </h2>
  <div>
    <label
      htmlFor="lat_lng"
    >
      Em que bairro será o serviço?
    </label>
    <br />
    <input
      name="lat_lng"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="name"
    >
      Nome
    </label>
    <br />
    <input
      name="name"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <br />
    <input
      name="email"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <label
      htmlFor="phone"
    >
      Celular
    </label>
    <br />
    <input
      name="phone"
      onChange={[Function]}
      placeholder={undefined}
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
  <button>
    Next
  </button>
</div>
`;
